crates/remote-prover-client/src/remote_prover/batch_prover.rs

```diff
@@ -90,17 +97,8 @@ impl RemoteBatchProver {
         proposed_batch: ProposedBatch,
     ) -> Result<ProvenBatch, RemoteProverClientError> {
         use miden_objects::utils::Serializable;
-        self.connect().await?;
 
-        let mut client = self
-            .client
-            .lock()
-            .await
-            .as_ref()
-            .ok_or_else(|| {
-                RemoteProverClientError::ConnectionFailed("client should be connected".into())
-            })?
-            .clone();
+        let mut client = self.connect().await?;
```

Comment:
Do we ever plan on replacing this client approach with the one through the builder?

As I understand it this pathway utilizes a client type that is not wrapped (like `BlockProducerStoreClient` for example).
```rust
    async fn connect(
        &self,
    ) -> Result<ApiClient<tonic::transport::Channel>, RemoteProverClientError> {
```